File: src/api_service/location.ts
import { ApiClient } from './client';

export interface Country {
  id: number;
  code: string;
  name: string;
}

export interface State {
  id: number;
  country_id: number;
  code: string;
  name: string;
}

export class LocationService {
  static async getCountries(): Promise<Country[]> {
    return ApiClient.get<Country[]>('/api/countries');
  }

  static async getStatesByCountry(countryId: number): Promise<State[]> {
    return ApiClient.get<State[]>(`/api/countries/${countryId}/states`);
  }
}

File: backend/internal/models/location.go
package models

type Country struct {
	ID   int16  `json:"id"`
	Code string `json:"code"`
	Name string `json:"name"`
}

type State struct {
	ID        int    `json:"id"`
	CountryID int16  `json:"country_id"`
	Code      string `json:"code"`
	Name      string `json:"name"`
}
